gulpfile.js

```diff
@@ -1,82 +1,123 @@
+  return gulp.src(src)
     .pipe(concat('app.js'))
     .pipe(uglify())
-    .pipe(gulp.dest('source/assets/js/'));
+    .pipe(gulp.dest(dest));
 });
 
-// minify & concatenate vendor scripts
-gulp.task('vendor', function() {
-  return gulp.src([
-    'source/_/js/vendor/*.js',
+// Minify vendor javascript
+gulp.task('uglifyvendor', function() {
+  const src = [
     'node_modules/highlightjs/highlight.pack.js',
     'node_modules/highlightjs-solidity/solidity.js',
-  ])
+  ]
+  const dest = distPath + 'assets/js/';
+  return gulp.src(src)
     .pipe(concat('vendor.js'))
     .pipe(uglify())
-    .pipe(gulp.dest('source/assets/js/'));
+    .pipe(gulp.dest(dest));
 });
 
-// compile sass and concatenate
-gulp.task('style', function() {
-  return gulp.src([
-    'source/_/sass/*.scss',
-    'node_modules/highlightjs/styles/default.css',
-  ])
-    .pipe(sass({
-      'sourcemap=none': true,
-      outputStyle: 'expanded'
-    }))
-    .pipe(concat('style.css'))
-    .pipe(gulp.dest('source/assets/css/'))
-    .pipe(notify({ message: 'Style task complete!' }));
+// Copy everything except the special folder _
+gulp.task('copyother', function() {
+  const f = ['**', '!**/*/', '!**/_/**', '!**/_'];
+  const f2 = function(vinyl) {
+    return !vinyl.isDirectory()
+  }
+  return gulp.src(srcPath + '**/*')
+    .pipe(filter(f))
+    .pipe(filter(f2))
+//    .pipe(gulp.symlink(distPath));
+    .pipe(gulp.dest(distPath));
 });
 
-// Browsersync
-gulp.task('browser-sync', function() {
-  browserSync.init(["source/**", "source/assets/css/*.css", "source/assets/js/*.js"], {
-    server: {
-      baseDir: "source"
-    }
-  });
-});
+// Complete regenerate the dist folder
+gulp.task('dist',
+  gulp.series('clean',
+    gulp.parallel(
+      'imagemin',
+      'minifyapp',
+      'minifyvendor',
+      'uglifyapp',
+      'uglifyvendor',
+      'copyother'
+    )
+  )
+);
 
-// watch these directories/files for changes
-gulp.task('watch', ['browser-sync'], function() {
-  // watch changes to all image, js & sass files within `_/` directory
-  gulp.watch('source/_/img/*', ['imagemin']),
-  gulp.watch('source/_/js/**/*.js', ['compress']),
-  gulp.watch('source/_/js/vendor/**/*.js', ['vendor']),
-  gulp.watch('source/_/sass/**/*.scss', ['style'])
+// Watch for changes
+gulp.task('watch', function() {
+  gulp.watch(srcPath + '_/img/**/*', gulp.task('imagemin'));
+  gulp.watch(srcPath + '_/sass/**/*.scss', gulp.task('minifyapp'));
+  gulp.watch(srcPath + 'node_modules/bulma/css/bulma.css', gulp.task('minifyvendor'));
+  gulp.watch(srcPath + '_/js/**/*.js', gulp.task('uglifyapp'));
+  gulp.watch('node_modules/**/*js', gulp.task('uglifyvendor'));
+  gulp.watch(srcPath, gulp.task('copyother'));
 });
 
-// tasks that run on 'gulp' command
-gulp.task('default', ['watch']);
-gulp.task('prod', function() {
-  gulp.src('source/index.html')
-  .pipe(gulp.dest('./'))
+// BrowserSync
+gulp.task('serve', gulp.series('watch', function() {
```

Comment:
I think this should be gulp.parallel, not gulp.series. series has 'watch' block serving the content